### PR TITLE
GH-1481 accept graph patterns in DELETE WHERE updates

### PR DESCRIPTION
**Pull request by deepali17043**
GitHub issue resolved: #1481 

Briefly describe the changes proposed in this PR:

We added a test that shows how the query described in #1481 can be run using SPARQLBuilder

---- 
PR Author Checklist: 

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [x] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.

### CODE REVIEW

File: contrib.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Kritika
+Deepali
+Anmol
```
[the remark this answers is not in the document]
Comment (deepali17043) [reply contrib.txt:3]: I'm sorry that was added as a check, I'll remove it and I've been signing all the commits but somehow there is some error, I'm looking into it and will update these according to the guidelines as soon as possible.

**Comment by deepali17043** [comment]
Yet to add ModifyQuery test

**Comment by 29kritika** [comment]
@jeenbroekstra we have resolved the changes requested but the same is not reflecting here in the checks. Please let us know if what we have done is alright or suggest if any change is requested.

File: core/sparqlbuilder/src/test/java/org/eclipse/rdf4j/sparqlbuilder/examples/updatespec/Section3Test.java [hunk 2]
```diff
@@ -342,4 +342,30 @@ public void example_14() {
...
 		p(Queries.ADD().fromDefault().to(iri("http://example.org/named")));
 	}
+
+	/**
+	 * DELETE { GRAPH ?g1 { ?subject <http://my-example.com/anyIRI/> ?object . ?subject ?predicate
+	 * <http://my-example.com/anyIRI/> . } } WHERE { GRAPH ?g1 { OPTIONAL { ?subject ?predicate
+	 * <http://my-example.com/anyIRI/> . } OPTIONAL { ?subject <http://my-example.com/anyIRI/> ?object . } } }
+	 */
+	@Test
+	public void example_issue_1481() {
```
Comment (jeenbroekstra) [inline core/sparqlbuilder/src/test/java/org/eclipse/rdf4j/sparqlbuilder/examples/updatespec/Section3Test.java:352]: This has been copied to a new class ModifyQueryTest, which is good, but you still need to remove it from Section3Test again.

Comment (deepali17043) [reply core/sparqlbuilder/src/test/java/org/eclipse/rdf4j/sparqlbuilder/examples/updatespec/Section3Test.java]: We have made removed the function from Section3Test

**Pull request merged**